File: mambo-protocol/src/main/scala/org/mambo/protocol/types/game/actions/fight/FightTemporaryBoostStateEffect.scala





















// Generated on 05/03/2013 10:35:02
// Scala version by Blackrush
package org.mambo.protocol.types

import org.mambo.core.io._
import org.mambo.protocol.{Type => BaseType, _}
import org.mambo.protocol.enums._

trait FightTemporaryBoostStateEffect extends FightTemporaryBoostEffect {
  override val typeId = 214

  protected trait Type extends FightTemporaryBoostEffect.Type with Serializable {
    override val typeId = FightTemporaryBoostStateEffect.typeId

    val stateId: Short

    override def serialize(buf: Buffer) {
      super.serialize(buf)

      buf.writeShort(stateId)
    }

  }

  protected def create(uid: Int, targetId: Int, turnDuration: Short, dispelable: Byte, spellId: Short, parentBoostUid: Int, delta: Short, stateId: Short): Target = throw new IllegalAccessException

  override def deserialize(buf: Buffer) = {
    val uid = buf.readInt()
    val targetId = buf.readInt()
    val turnDuration = buf.readShort()
    val dispelable = buf.readByte()
    val spellId = buf.readShort()
    val parentBoostUid = buf.readInt()
    val delta = buf.readShort()
    val stateId = buf.readShort()

    create(uid, targetId, turnDuration, dispelable, spellId, parentBoostUid, delta, stateId)
  }

}

object FightTemporaryBoostStateEffect extends FightTemporaryBoostStateEffect {
  case class TypeImpl(uid: Int, targetId: Int, turnDuration: Short, dispelable: Byte, spellId: Short, parentBoostUid: Int, delta: Short, stateId: Short) extends Type
  override def create(uid: Int, targetId: Int, turnDuration: Short, dispelable: Byte, spellId: Short, parentBoostUid: Int, delta: Short, stateId: Short) = TypeImpl(uid, targetId, turnDuration, dispelable, spellId, parentBoostUid, delta, stateId)

  def unapply(msg: TypeImpl) = TypeImpl.unapply(msg)
  def apply(uid: Int, targetId: Int, turnDuration: Short, dispelable: Byte, spellId: Short, parentBoostUid: Int, delta: Short, stateId: Short) = create(uid, targetId, turnDuration, dispelable, spellId, parentBoostUid, delta, stateId)

  type Target = TypeImpl

}

File: mambo-protocol/src/main/java/org/mambo/protocol/types/FightTemporaryBoostStateEffect.java


// Generated on 05/08/2013 19:38:00
package org.mambo.protocol.types;

import java.util.*;
import org.mambo.protocol.enums.*;
import org.mambo.protocol.*;
import org.mambo.core.io.*;

public class FightTemporaryBoostStateEffect extends FightTemporaryBoostEffect {
    public static final short TYPE_ID = 214;
    
    @Override
    public short getTypeId() {
        return TYPE_ID;
    }
    
    public short stateId;
    
    public FightTemporaryBoostStateEffect() { }
    
    public FightTemporaryBoostStateEffect(int uid, int targetId, short turnDuration, byte dispelable, short spellId, int parentBoostUid, short delta, short stateId) {
        super(uid, targetId, turnDuration, dispelable, spellId, parentBoostUid, delta);
        this.stateId = stateId;
    }
    
    @Override
    public void serialize(Buffer buf) {
        super.serialize(buf);
        buf.writeShort(stateId);
    }
    
    @Override
    public void deserialize(Buffer buf) {
        super.deserialize(buf);
        stateId = buf.readShort();
    }
    
}
